src/components/hoc-helpers/withHeaderName.js

```diff
@@ -0,0 +1,25 @@
+import React, { Component } from 'react';
+
+const withHeaderName = (View) => {
+  return class extends Component {
+    constructor(props) {
+      super(props);
+      this.state = {
+          controlHeader: '',
+          setControlHeader: this.setControlHeader
+        };
+    }
+  
+    setControlHeader = (event, controlHeader) => {
+      this.setState({
+        controlHeader,
+      });
+    };
+  
+    render() {
+      return (<View {...this.props} {...this.state} />);
```

Comment:
you pass all the props you get from outside, it's ok
then you need to pass through setControlHeader and control header, right?
so I would suggest to remove setControlHeader from the state(because it's not a kind of state) and do something like that
```suggestion
      return (<View {...this.props} controlHeader={this.state.controlHeader} onHeaderChange={this.setControlHeader} />);
```

make sense?

Reply:
Yep, makes sense indeed.